lib/vnet/osconfig_provider.go

```diff
@@ -89,33 +91,34 @@ func (p *osConfigProvider) setV4IPsFromFirstCIDR(cidrRange string) error {
 		// Only set these once.
 		return nil
 	}
-	tunIPv4, dnsIPv4, err := ipsForCIDR(cidrRange)
+	tunIPv4, tunIPv4Net, dnsIPv4, err := ipsForCIDR(cidrRange)
 	if err != nil {
 		return trace.Wrap(err, "setting TUN IPv4 address for range %s", cidrRange)
 	}
 	if err := p.cfg.addDNSAddress(dnsIPv4); err != nil {
 		return trace.Wrap(err, "adding IPv4 DNS server at %s", dnsIPv4.String())
 	}
 	p.tunIPv4 = tunIPv4.String()
+	p.tunIPv4Net = tunIPv4Net
 	p.dnsAddrs = append(p.dnsAddrs, dnsIPv4.String())
 	return nil
 }
 
 // ipsForCIDR returns the V4 IPs to assign to the interface and use for DNS in
 // cidrRange.
-func ipsForCIDR(cidrRange string) (tunIP net.IP, dnsIP net.IP, err error) {
-	_, ipnet, err := net.ParseCIDR(cidrRange)
+func ipsForCIDR(cidrRange string) (tunIP net.IP, tunIPNet *net.IPNet, dnsIP net.IP, err error) {
+	_, tunIPNet, err = net.ParseCIDR(cidrRange)
```

Comment:
The comment below still refers to the original `ipnet` variable name. Please update.